src/ansys/fluent/core/codegen/settingsgen.py

```diff
@@ -50,486 +55,188 @@ def _gethash(obj_info):
-                )
-                f.write(f"{istr1})\n\n")
-
-            child_aliases = getattr(cls, "_child_aliases", None)
-            if child_aliases:
-                f.write(f"{istr1}_child_aliases = dict(\n")
-                f.writelines([f'{istr2}{k}="{v}",\n' for k, v in child_aliases.items()])
-                f.write(f"{istr1})\n\n")
-
-            # write object type
-            child_object_type = getattr(cls, "child_object_type", None)
-            if child_object_type:
-                f.write(f"{istr1}child_object_type: {pchild_name} = {pchild_name}\n")
-                f.write(f'{istr1}"""\n')
-                f.write(f"{istr1}child_object_type of {cls_name}.")
-                f.write(f'\n{istr1}"""\n')
-                if stub_f:
-                    stub_f.write(f"{istr1}child_object_type: {pchild_name} = ...\n")
-
-            return_type = getattr(cls, "return_type", None)
-            if return_type:
-                f.write(f'{istr1}return_type = "{return_type}"\n')
-                if stub_f:
-                    stub_f.write(f"{istr1}return_type = ...\n")
-            if stub_f:
-                stub_f.close()
-
-
-def _populate_init(parent_dir, hash):
-    file_name = os.path.normpath(os.path.join(parent_dir, "__init__.py"))
-    with open(file_name, "w") as f:
-        f.write("#\n")
-        f.write("# This is an auto-generated file.  DO NOT EDIT!\n")
-        f.write("#\n")
-        f.write("\n")
-        f.write(f'"""A package providing Fluent\'s Settings Objects in Python."""')
-        f.write("\n")
-        f.write("from ansys.fluent.core.solver.flobject import *\n\n")
-        f.write(f'SHASH = "{hash}"\n')
-        f.write(f"from .{root_class_path} import root")
-
-
-def generate(version, static_infos: dict):
-    """Generate settings API classes."""
-    parent_dir = (pyfluent.CODEGEN_OUTDIR / "solver" / f"settings_{version}").resolve()
+def _write_function_stub(name, data, s_stub):
+    s_stub.write(f"    def {name}(self, ")
+    for arg_name in data["argument_names"]:
+        arg_type = _arg_type_strings[data["child_classes"][arg_name]["bases"][0]]
+        s_stub.write(f"{arg_name}: {arg_type}, ")
+    s_stub.write(f"):")
+    # TODO: add return type
+    doc = data["doc"]
+    doc = doc.replace("\n", "\n        ")
+    s_stub.write('        """\n')
+    s_stub.write(f"        {doc}")
+    s_stub.write('        """\n')
+
+
+def _write_data(cls_name: str, python_name: str, data: dict, f: IO, f_stub: IO | None):
+    s = StringIO()
+    s_stub = StringIO()
+    bases = ", ".join(data["bases"])
+    cls_def = f"class {cls_name}({bases}):\n"
+    s.write(cls_def)
+    s_stub.write(cls_def)
+    doc = data["doc"]
+    doc = doc.replace("\n", "\n    ")
+    s.write('    """\n')
+    s.write(f"    {doc}\n")
+    s.write('    """\n')
+    s.write(f"    version = {data['version']!r}\n")
+    s.write(f"    fluent_name = {data['fluent_name']!r}\n")
+    s.write(f"    _python_name = {python_name!r}\n")
```

Comment:
This is a new class-level attribute. One change with the new mechanism is that class names are indexed (like `file_name_5`) because they are all defined within the same namespace. `_python_name` preserves the original class name. It is used for `python_name` and `python_path` properties of setting objects.